lib/errorlogger.cpp

```diff
@@ -1045,12 +1045,9 @@ std::string getGuideline(const std::string &errId, ReportType reportType,
             guideline = errId.substr(errId.rfind('-') + 1);
         break;
     case ReportType::misraCpp2008:
-        if (errId.rfind("misra-cpp-2008-", 0) == 0)
-            guideline = errId.substr(15);
-        break;
     case ReportType::misraCpp2023:
-        if (errId.rfind("misra-cpp-2023-", 0) == 0)
-            guideline = errId.substr(15);
+        if (errId.rfind("misra-cpp-20-", 0) == 0 || errId.rfind("premium-misra-cpp-20", 0) == 0)
+            guideline = errId.substr(errId.rfind('-') + 1);
```

Comment:
The Misra C++ 2008 uses `-` instead of `.` Example:
premium-misra-cpp-2008-1-2-3